src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/Descriptor.php

```diff
@@ -293,21 +291,13 @@ protected function sortServiceIds(array $serviceIds)
     public static function getClassDescription(string $class, string &$resolvedClass = null): string
     {
         $resolvedClass = $class;
-
-        if (!interface_exists(DocBlockFactoryInterface::class)) {
-            return '';
-        }
-
         try {
             $r = new \ReflectionClass($class);
             $resolvedClass = $r->name;
-
-            if ($docComment = $r->getDocComment()) {
-                return DocBlockFactory::createInstance()
-                    ->create($docComment)
-                    ->getSummary();
+            if (('' !== $docComment = $r->getDocComment()) && false !== preg_match('~^\s*+/\*\*\s++\*\s*+(?<!@)(\w.++)~', $docComment, $matches)) {
```

Comment:
- getDocComment can return false - `if ($docComment = $r->getDocComment()) {` was really fine let's keep it
- descriptions can span several lines - the current logic doesn't account for them + we should clean noise `*` & `\n` between lines to always return one single line.

Reply:
Here's the new regex: https://regex101.com/r/2bXDAG/1 :)

you'll need to trim and replace `"\n * "` still in group1 from match1

Reply:
I'd doit differently: first use preg_split to get the description, then a preg_replace to clean it.

Reply:
@nicolas-grekas I'm +1 for using just the first line of the description. Otherwise we'll endup with descriptions like
` * UrlGeneratorInterface is the interface that all URL generator classes must implement.
 *
 * The constants in this interface define the different types of resource references that
 * are declared in RFC 3986: http://tools.ietf.org/html/rfc3986
 * We are using the term "URL" instead of "URI" as this is more common in web applications
 * and we do not need to distinguish them as the difference is mostly semantical and
 * less technical. Generating URIs, i.e. representation-independent resource identifiers,
 * is also possible.` for UrlGeneratorInterface for ex.

Reply:
The short description is delimited by 2 consecutive newlines. URL generator is OK :)

https://github.com/symfony/symfony/blob/df4ad4e7d4997a2f186c14312a250f2fca2c1b03/src/Symfony/Component/Routing/Generator/UrlGeneratorInterface.php#L20-L21

Reply:
@ro0NL Oh now I get what @nicolas-grekas meant :) thx for clarifying